loader.rb

```diff
@@ -0,0 +1,60 @@
+require 'json'
+require './book'
+require './person'
+require './rentals'
+class Loader
+  def initialize
+    load_data
+  end
+
+  def load_data
+    @books = load_books
+    @rentals = load_rentals
+    @people = load_people
+  end
+
+  def load_books
+    return [] unless File.exist?('storage/books.json')
+
+    books_json = File.read('storage/books.json')
+    return [] if books_json.empty?
+
+    books_data = JSON.parse(books_json)
+    books_data.map { |data| Book.new(data['title'], data['author']) }
+  end
+
+  def load_rentals
+    return [] unless File.exist?('storage/rentals.json')
+
+    rentals_json = File.read('storage/rentals.json')
+    return [] if rentals_json.empty?
+
+    rentals_data = JSON.parse(rentals_json)
+    rentals_data.map do |data|
+      book_data = data['book']
+      person_data = data['person']
+      book = @books.find { |b| b.title == book_data['title'] && b.author == book_data['author'] }
+      person = create_person_from_data(person_data)
+      Rentals.new(data['date'], person, book)
+    end
+  end
+
+  def load_people
+    return [] unless File.exist?('storage/people.json')
+
+    people_json = File.read('storage/people.json')
+    return [] if people_json.empty?
+
+    people_data = JSON.parse(people_json)
+    people_data.map { |data| create_person_from_data(data) }
+  end
+
+  def create_person_from_data(data)
+    if data['specialization']
+      Teacher.new(data['age'], data['name'], data['specialization'])
+    else
+      Student.new(data['age'], data['name'])
+    end
+  end
```

Comment:
On startup of your app, the method `List all rentals` is not working as expected and your app is not loading the list of rentals
![image](https://github.com/ahmed025566/oop_school_libarry/assets/78251927/d114a131-681d-42b6-9ac3-ac7badfcc589)

database for rentals
![image](https://github.com/ahmed025566/oop_school_libarry/assets/78251927/668aa2a0-79e3-4ea7-990a-4e5aa9153bdb)

This is because the newly generated `ID` is different from the `ID` in the database.

However, since we are fetching the list of rentals using the saved `ID`'s, kindly ensure that the `ID`s are all the same

e.g
```ruby
    person = if record['type'] == 'Student'
                 Student.new(
                   nil,
                   record['age'],
                   record['name'],
                   parent_permission: record['parent_permission']
                 )
               else
                 Teacher.new('', record['age'], record['name'])
               end
      person.id = record['id']
      people << person
    end
```

Your app
`ID` saved != new `ID`

![image](https://github.com/ahmed025566/oop_school_libarry/assets/78251927/e71f1299-2b9c-4c8f-8495-e6aafd89db6f)

![image](https://github.com/ahmed025566/oop_school_libarry/assets/78251927/75d5d82f-be2e-431f-967f-de9b223ebed1)